ocaml/parsing/printast.ml

```diff
@@ -911,21 +911,22 @@ and longident_x_pattern i ppf (li, p) =
   line i ppf "%a\n" fmt_longident_loc li;
   pattern (i+1) ppf p;
 
-and case i ppf {pc_lhs; pc_guard; pc_rhs} =
+and case i ppf { pc_lhs; pc_rhs } =
   line i ppf "<case>\n";
   pattern (i+1) ppf pc_lhs;
-  begin match pc_guard with
-  | None -> ()
-  | Some g -> line (i+1) ppf "<when>\n"; guard (i + 2) ppf g
-  end;
-  expression (i+1) ppf pc_rhs;
-
-and guard i ppf = function
-  | Guard_predicate e -> expression i ppf e
-  | Guard_pattern { pgp_scrutinee = e; pgp_pattern = pat } ->
-      expression i ppf e;
-      line i ppf "<match>\n";
-      pattern (i + 1) ppf pat
+  case_rhs (i+1) ppf pc_rhs;
+
+and case_rhs i ppf = function
+  | Psimple_rhs e -> expression i ppf e
+  | Pboolean_guarded_rhs { pbg_guard; pbg_rhs } ->
+      line i ppf "<when>\n";
+      expression (i + 1) ppf pbg_guard;
+      expression i ppf pbg_rhs
+  | Ppattern_guarded_rhs { ppg_scrutinee; ppg_cases } ->
+      line i ppf "when\n";
+      expression (i + 1) ppf ppg_scrutinee;
+      line (i + 1) ppf "match\n";
```

Comment:
The printing here seems a bit different than the style in the rest of the file. Usually, the less-indented line at the top is kind of like a variant constructor in that it identifies what data will follow. I think I'd prefer `<when-pattern>` to the current state of `when` followed by `match` at some later line.

Reply:
That makes sense, done!